src/java.base/share/classes/java/lang/System.java

```diff
@@ -2594,6 +2594,14 @@ public void inflateBytesToChars(byte[] src, int srcOff, char[] dst, int dstOff,
                 StringLatin1.inflate(src, srcOff, dst, dstOff, len);
             }
 
+            public boolean isLatin1(String s) {
+                return s.coder() == String.LATIN1;
```

Comment:
I recall JEP 254 authors made a point that the test for `COMPACT_STRINGS` in `String#isLatin1()` helps JIT do better dead code elimination when running with `-XX:-CompactStrings`. Also: no logic in access bridges. :)  
```suggestion
                return s.isLatin1();
```